### PR TITLE
CRM-10551: Allow deletion of blanked location-based fields edited via a profile

### PR DESCRIPTION
**Pull request by JKingsnorth**

### CODE REVIEW

**Comment by davecivicrm** [comment]
ok to test

**Comment by JKingsnorth** [comment]
The last commit removes the whitespace from the blank line, so it should pass the test now.

**Comment by JKingsnorth** [comment]
Woohoo, tests passed.

**Comment by JKingsnorth** [comment]
Sorry for the test bot spam. I haven't managed to get my local environment set up properly yet to run the tests. I'm working on it though.

File: CRM/Core/BAO/Block.php [hunk 1]
```diff
@@ -239,153 +245,134 @@ public static function create($blockName, &$params, $entity = NULL, $contactId =
...
-                $resetPrimaryId = $blockId;
-                break;
-              }
-            }
-          }
-          if ($resetPrimaryId) {
-            $baoString = 'CRM_Core_BAO_' . $blockName;
-            $block = new $baoString();
+    // For each submitted value
+    foreach ($submitted_values as $count => $submitted_value) {
```
Comment (kurund) [inline CRM/Core/BAO/Block.php]: Can you please update the code to use camecase convention.  eg: $submittedValues instead of $submitted_values

Comment (JKingsnorth) [inline CRM/Core/BAO/Block.php]: Oh bother, I was following Drupal's coding standards. That's frustrating. OK I'll take a look at this.

Comment (colemanw) [inline CRM/Core/BAO/Block.php]: To be fair, I don't think Civi's coding standards take a hard line on this. But it's true that we use camelCase far more often than underscores and consistency is nice.

Comment (JKingsnorth) [inline CRM/Core/BAO/Block.php]: No worries coleman, consistency is king! It is mentioned here: http://wiki.civicrm.org/confluence/display/CRMDOC/PHP+Code+and+Inline+Documentation

**Comment by JKingsnorth** [comment]
camelCase variable names are now in place

**Comment by davecivicrm** [comment]
Ran through the manual test cases including the prior 'failure' case and all scenarios are behaving as expected.

**Pull request merged**